static/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -414,14 +415,29 @@ function getPayload<T, P>(props: Props<T, P>) {
   return payload;
 }
 
+const getDateComponentCacheKey = (statsPeriod: string | undefined): string => {
+  if (!statsPeriod) {
+    return '';
+  }
+  const unit = statsPeriod.slice(-1);
+  // If the unit is in days, then the cache invalidates after an hour, otherwise 5 minutes
+  const date = moment();
+  date.set({second: 0, millisecond: 0});
+  if (unit === 'd' || unit === 'm' || unit === 'w') {
+    return date.endOf('hour').format();
+  }
+  const remainder = 5 - (date.minute() % 5);
+  return date.add(remainder, 'minutes').format();
+};
+
 export function useGenericDiscoverQuery<T, P>(props: Props<T, P>) {
   const api = useApi();
   const {orgSlug, route, options} = props;
   const url = `/organizations/${orgSlug}/${route}/`;
   const apiPayload = getPayload<T, P>(props);
 
   const res = useQuery<[T, string | undefined, ResponseMeta<T> | undefined], QueryError>(
-    [route, apiPayload],
+    [route, apiPayload, getDateComponentCacheKey(apiPayload.statsPeriod)],
```

Comment:
Putting this here does eliminate some control over when cache should be invalidated, maybe instead a prop containing `additionalCacheKey` information would be better here. For example the response that contains the span op and the query itself shouldn't change regardless of stats period.